src/components/Mentions/MentionButton.tsx

```diff
@@ -0,0 +1,116 @@
+import { List } from 'immutable';
+
+import { FC, useMemo, useState } from 'react';
+import { I18nextProvider } from 'react-i18next';
+import { QueryObserverResult } from 'react-query';
+
+import { Badge, IconButton, makeStyles } from '@material-ui/core';
+import { Notifications } from '@material-ui/icons';
+
+import {
+  MemberMentionsRecord,
+  MemberRecord,
+} from '@graasp/query-client/dist/src/types';
+import { MentionStatus } from '@graasp/sdk';
+import buildI18n, { CHATBOX, langs, namespaces } from '@graasp/translations';
+
+import { mentionButtonCypress } from '../../config/selectors';
+import MentionsDialog from './MentionsDialog';
+import MentionsTable from './MentionsTable';
+
+type Props = {
+  color?: 'primary' | 'secondary';
+  lang?: string;
+  useMentions: (
+    options?: { getUpdates?: boolean | undefined } | undefined,
+  ) => QueryObserverResult<MemberMentionsRecord>;
+  useMembers: (memberIds: string[]) => QueryObserverResult<List<MemberRecord>>;
+  patchMentionFunction: (args: { id: string; status: string }) => void;
+  deleteMentionFunction: (id: string) => void;
+  clearAllMentionsFunction: () => void;
+};
+
+const MentionButton: FC<Props> = ({
+  color = 'primary',
+  lang = langs.en,
+  useMentions,
+  useMembers,
+  patchMentionFunction,
+  deleteMentionFunction,
+  clearAllMentionsFunction,
+}) => {
+  const i18n = useMemo(() => {
+    const i18nInstance = buildI18n(namespaces.chatbox);
+    i18nInstance.changeLanguage(lang);
+    return i18nInstance;
+  }, [lang]);
```

Comment:
Isn't it possible to have a separate file to instantiate it, then you change the lang in a use effect?

Reply:
I think the issue is that we are dealing with a component here ... so i am not sure that using a different file to just put `buildI18n(namespaces.chatbox)` ?